src/sync/sync_permission.hpp

```diff
@@ -150,11 +156,19 @@ class Permissions {
     using ConfigMaker = std::function<Realm::Config(std::shared_ptr<SyncUser>, std::string url)>;
 
     // Callback used to asynchronously vend a `PermissionResults` object.
+    // DEPRECATED. Will be replaced by RawPermissionResultsCallback in the future.
     using PermissionResultsCallback = std::function<void(std::unique_ptr<PermissionResults>, std::exception_ptr)>;
 
+    // Callback used to asynchronously vend permissions results.
+    using RawPermissionResultsCallback = std::function<void(Results, std::exception_ptr)>;
```

Comment:
Our existing `get_permissions()` static API gets the raw `realm::Results`, wraps it in a `PermissionResults`, and then returns it in that callback.

I don't want that wrapping step anymore; I just want the raw `realm::Results`, but I don't want to break the other bindings, hence this new callback and new static method that work the same as the existing ones.